core/sql/generator/GenRelUpdate.cpp

```diff
@@ -1718,19 +1806,56 @@ short HbaseUpdate::codeGen(Generator * generator)
     }
   else if (getIndexDesc()->isClusteringIndex() && getCheckConstraints().entries())
     {
-      GenAssert(FALSE, "Should not reach here. This update should have been transformed to delete/insert");
-      // To be uncommented when TRAFODION-1610 is implemented
-      // Need to generate insConstraintExpr also
-/*
+      // Generate the update and insert constraint check expressions
+
+      // The attributes for the columns referenced in the constraint expressions
+      // refer to the source values of the columns. We want to evaluate the
+      // constraints aganst the target values, though. So, there is some
+      // Attributes gymnastics that has to happen to generate them.
```

Comment:
Are the Attributes gymnastics gone with the new code? Is it now all couch potatoes?

Reply:
Oops... I should have deleted the comments from 1811-1814

Reply:
Or at least edited them